### ISSUE TITLE
Is this a `consistent-function-scoping` bug?

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#391]
I was working on #387 , and found our code 

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/92f2da10fd97841e8d674c17f397c39c24bf9fe2/test/integration/test.js#L57-L65

is reported by `consistent-function-scoping`.

This code looks good to me, is this a bug or did I miss something?

/cc @MrHen

**Comment by MrHen** [comment]
Definitely looks similar to #372 but I'll keep the use case around just to be sure.

**Comment by amankkg** [comment]
Seems like another case for an issue:
```js
export const canStepForward = ([X, Y]) => ([x, y]) => direction => {
  switch (direction) {
    case 0:
      return y !== 0
    case 1:
      return x !== X - 1
    case 2:
      return y !== Y - 1
    case 3:
      return x !== 0
    default:
      throw new Error('unknown direction')
  }
}
```

> `1:43`  `error`  Move arrow function to the outer scope  `unicorn/consistent-function-scoping`

`1:43` points to second function, e.g. `([x, y]) => direction =>`

Seems like currying is not handled properly because refactoring to this removes ESLint error:
```js
export const canStepForward = ([X, Y]) => {
  return ([x, y]) => direction => {
    switch (direction) { ... }
  }
}
```
